core/src/main/java/org/mule/runtime/core/execution/ExceptionToMessagingExceptionExecutionInterceptor.java

```diff
@@ -42,74 +39,30 @@ public Event execute(Processor messageProcessor, Event event) throws MessagingEx
       if (exception instanceof MessagingException) {
         //Use same exception, but make sure whether a new error is needed
         messagingException = (MessagingException) exception;
-        Throwable causeException = exception.getCause() != null ? exception.getCause() : exception;
-        Optional<Error> error = messagingException.getEvent().getError();
         // TODO - MULE-10266 - Once we remove the usage of MessagingException from within the mule component we can get rid of the
         // messagingException.causedExactlyBy(..) condition.
-        if (!error.isPresent() || !error.get().getCause().equals(causeException)
-            || !messagingException.causedExactlyBy(error.get().getCause().getClass())) {
-          ErrorType errorType = getErrorTypeFromFailingProcessor(messageProcessor, causeException);
-          event = Event.builder(messagingException.getEvent())
-              .error(ErrorBuilder.builder(causeException).errorType(errorType).build()).build();
-          messagingException.setProcessedEvent(event);
-        }
+        event = createErrorEvent(event, messageProcessor, messagingException, muleContext);
```

Comment:
Discussed yes, this is just refactor of code out of this interceptor and to somewhere common.